mala/targets/ldos.py

```diff
@@ -520,6 +547,92 @@ def read_from_array(self, array, units="1/(eV*A^3)"):
         self.local_density_of_states = array * \
                                        self.convert_units(1, in_units=units)
 
+    def read_from_hdf5(self, path, units=None):
+        """
+        Read the LDOS data from an OpenPMD HDF5 file.
+
+        The LDOS data will be saved within the LDOS object.
+
+        Parameters
+        ----------
+        path : string
+            Name of the HDF5 file.
+
+        units : string
+            Units the LDOS data is saved in. Currently ignored
+            in the OpenPMD formalism.
+        """
+        series = io.Series(path, io.Access.read_only)
+
+        # Check if this actually MALA compatible data.
+        if series.get_attribute("is_mala_data") != 1:
+            raise Exception("Non-MALA data detected, cannot work with this "
+                            "data.")
+
+        # A bit clanky, but this way only the FIRST iteration is loaded,
+        # which is what we need for loading from a single file that
+        # may be whatever iteration in its series.
+        for current_iteration_index in series.iterations:
+            current_iteration = series.iterations[current_iteration_index]
+            ldos_mesh = current_iteration.meshes["LDOS"]
+            break
+
+        self.local_density_of_states = \
+            np.zeros((ldos_mesh["0"].shape[0], ldos_mesh["0"].shape[1],
+                     ldos_mesh["0"].shape[2], len(ldos_mesh)),
+                     dtype=ldos_mesh["0"].dtype)
+
+        # TODO: This is currently inefficient.
+        # But if I register the chunks to individual indices i of the
+        # actual LDOS object, then something goes wrong during the flushing,
+        # so I flush multiple times...
+        for i in range(0, len(ldos_mesh)):
+
+            # TODO: Fix this check.
+            # This check checks whether the correct units have been used.
+            # It's not entirely trivial to check this, since OpenPMD works
+            # in SI units, but MALA in eV/Ang units. So the correct factor
+            # is not going to be 1.0 - it's going to be (m**3)*J, and due to
+            # machine precision, any test to this effect may be slightly off.
+            # Since there is no sensible reason to not save new MALA data
+            # with the wrong units, for now we just check and complain.
+            # Eventually we may have to implement unit conversion, I guess.
+            if not np.isclose(ldos_mesh[str(i)].unit_SI, ((m**3)*J)):
+                raise Exception("MALA currently cannot operate with HDF5 "
+                                "files with non-MALA units.")
+
+            temp_ldos = ldos_mesh[str(i)].load_chunk()
+            series.flush()
+            self.local_density_of_states[:, :, :, i] = temp_ldos.copy()
```

Comment:
I think that making this more efficient without error is a matter of picking the right variant of load_chunk(). 
My current hypothesis:
**What you wanted to do** Load data into an existing buffer
**What you accidentally did** Replace the existing buffer with the new buffer, created by `load_chunk`

There exists also a version of `load_chunk` that works with existing buffers, we'll have to carefully pick the right one here.